models/migrations/v78.go

```diff
@@ -37,6 +38,17 @@ func renameRepoIsBareToIsEmpty(x *xorm.Engine) error {
 		return fmt.Errorf("Drop index failed: %v", err)
 	}
 
+	if err = sess.Commit(); err != nil {
+		return err
+	}
+
+	// Then reset the values
+	sess = x.NewSession()
+	defer sess.Close()
```

Comment:
I think this two lines is unnecessary but xorm should work.

Reply:
Hiya @lunny, dropping the index and sync2 have to be in different transactions on sqlite as the schema becomes locked by sync2 which conflicts with the index drop. Therefore the original single transaction fails on sqlite and is rolled back.

Reply:
@zeripath I know. I mean
```Go
sess := x.NewSession()
defer sess.Close()

sess.Begin()
sess.Commit()

sess.Begin()
sess.Commit()
```
The second `NewSession` is uncessary.